composer/datasets/cifar.py

```diff
@@ -195,8 +195,8 @@ class CIFARWebDatasetHparams(WebDatasetHparams):
     height: int = hp.optional('Image height', default=32)
     width: int = hp.optional('Image width', default=32)
     n_classes: int = hp.optional('Number of output classes', default=0)
-    channel_means: List[float] = hp.optional('Mean per image channel', default=(0, 0, 0))
-    channel_stds: List[float] = hp.optional('Std per image channel', default=(0, 0, 0))
+    channel_means: List[float] = hp.optional('Mean per image channel', default_factory=lambda: [0, 0, 0])
+    channel_stds: List[float] = hp.optional('Std per image channel', default_factory=lambda: [0, 0, 0])
```

Comment:
this seems odd to me, the type is specified as a list but default is a lamba function. is there a yahp specific reason why this needs to be this way?

Reply:
Yeah, this is a bit unintuitive, but here's what's happening:

1. YAHP does not support tuples or generic sequences; only lists (at least not yet), so the type has to be `List[float]`
2. `hp.optional` now needs to return a list type instead of a tuple
3. Because lists are mutable, it is necessary to provide a factory, so each instance of the dataclass will get a different instance of the list. See https://docs.python.org/3/library/dataclasses.html#dataclasses.field